### DISCUSSION TITLE
Feature Request: Partition By Enhancement - Exclude Partition Columns From Written File

### DISCUSSION THREAD

**Discussion by brianwyka** [duckdb/duckdb#8610] [Ideas]
It would be nice to be able to set an option on the `COPY` statement to exclude columns which are part of the `partition_by` option from being written to the resulting file and schema.

For example:

ducks table:
| id                | species | name |
| ----------  | -------- | ----- |
| 1                 | wood      | quick |
| 2                | mallard   | quack |

```sql
COPY ducks TO 'data/' (PARTITION_BY ('species') EXCLUDE_PARTITION_COLUMNS_FROM_FILE TRUE);
```

The resulting files would have columns `id` and `name`
- `/data/species=wood/data_0.csv`
- `/data/species=mallard/data_0.csv`

In tools like Athena, when hive-style partitions are used in S3, Athena derives the column from the partition, instead of needing it in the file, and if its written into the file and included in the schema, then its duplicated.

I originally tried with this workaround, but query failed due to `species` column not in the result set:

```sql
COPY (SELECT * EXCLUDE ('species') FROM ducks) TO 'data/' (PARTITION_BY ('species'));
```

**Comment by m-richards** [comment]
I'd similarly be interested in the capability but for parquet files. I have been looking at duckdb as an alternative for writing hive partitioned parquets and noticed it is significantly faster than pyarrow at writing parquets. However the system I am feeding the parquet to cannot handle columns appearing in both the hive file path and the file.

**Comment by brianwyka** [comment]
Looks like this is handled now according to the documentation here: https://duckdb.org/docs/sql/statements/copy

See `WRITE_PARTITION_COLUMNS` option.
